src/api/app/controllers/concerns/webui/rescue_handler.rb

```diff
@@ -52,5 +43,55 @@ class MissingParameterError < RuntimeError; end
       logger.debug "#{exception.class.name} #{exception.message} #{exception.backtrace.join('\n')}"
       render file: Rails.root.join('public/404'), status: 404, layout: false, formats: [:html]
     end
+
+    private
+
+    UNAUTHORIZED_MESSAGE = 'Sorry, you are not authorized to perform this action.'.freeze
+    def unauthorized_message(exception)
+      pundit_action = action_for_query(exception.query)
+
+      if exception.reason
+        message_for_reason(exception.reason)
+      elsif pundit_action && exception.record
+        "Sorry, you are not authorized to #{pundit_action} this #{exception.record.class}."
+      else
+        UNAUTHORIZED_MESSAGE
+      end
+    end
+
+    def action_for_query(query) # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
The `Metrics/CyclomaticComplexity` appeared after moving the code. 🤷‍♂️

Reply:
i ignored the codeclimate issue. We can deal with that later.